app/api/og-image/cast/[username]/[hash]/[[...rank]]/route.tsx

```diff
@@ -0,0 +1,198 @@
+                {metadata?.trackName}
+              </p>
+              <p tw="text-xl m-0 text-[#949494]">
+                {replaceSpecialCharacters(metadata?.artistName)}
+              </p>
+            </div>
+            <div tw="flex flex-col m-0 bg-[#F6F6F6] rounded-2xl w-[240px] h-[145px] items-center justify-center ml-[40px] absolute left-1/3 top-5 ">
+              <div tw="flex items-center m-0 w-[68%]">
+                <p tw="m-0 text-4xl text-[#333536]" style={{ fontWeight: 600 }}>
+                  {points}
+                </p>
+                <img
+                  src="https://i.imgur.com/Xa4LjYA.jpeg"
+                  alt=""
+                  width={100}
+                  height={100}
+                  loading="lazy"
+                  tw="rounded-full ml-2 w-16 h-16"
+                  style={{ imageRendering: 'pixelated' }}
+                />
+              </div>
+              <p tw="m-0 text-xl text-[#949494]" style={{ fontWeight: 400 }}>
+                Notes Collected
+              </p>
+            </div>
+          </div>
+          <div tw="flex flex-col mt-10 ml-1 ">
+            <p tw="text-[#949494] text-xl" style={{ fontWeight: 400 }}>
+              Posted By
+            </p>
+            <div tw="flex">
+              <div tw="flex items-center rounded-full">
+                <img
+                  tw="rounded-full"
+                  src={profilePfp}
+                  alt="warpcast"
+                  width={50}
+                  height={50}
+                  loading="lazy"
+                />
+              </div>
+              {rank && (
+                <div tw="flex items-center justify-center bg-[#F6F6F6] rounded-full w-[100px] ml-5">
+                  <p tw="text-xl ml-2 m-0 text-[#333536] " style={{ fontWeight: 600 }}>
+                    #{rank}
+                  </p>
+                </div>
+              )}
+              <div tw="flex items-center justify-center bg-[#EEE4FE] rounded-full max-w-[220px] w-fit px-5 h-12 ml-5">
+                <img
+                  src="https://i.imgur.com/JXN6jYv.png"
+                  alt="warpcast"
+                  width={28}
+                  height={24}
+                  loading="lazy"
+                />
+                <p tw="text-[#8b49f7] text-xl ml-2" style={{ fontWeight: 600 }}>
+                  {encodedUsername}
+                </p>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div tw="flex absolute bottom-4 left-[47%] justify-center">
+          <p tw="text-[12px]" style={{ fontWeight: 400 }}>
+            sonata.tips
+          </p>
+        </div>
+      </div>
+    ),
+    {
+      width: 1200,
+      height: 630,
+      fonts: [
+        {
+          name: 'Sora',
+          data: await soraNormal,
+          weight: 400,
+        },
+        {
+          name: 'Sora',
+          data: await soraSemiBold,
+          weight: 600,
+        },
+      ],
+    },
+  );
+}
```

Comment:
file too long
- rule: files should be a `less than 100 lines of code`
- actual: @TahirHameed74 file has ~200 lines of code
- suggestion: break this down into smaller libs / reusable components